packages/types/index.d.ts

```diff
@@ -4,15 +4,19 @@ import type {
   InternalAxiosRequestConfig,
 } from 'axios'
 
+// FIXME (ogp-weeloong): temporarily importing these to migrate from IJSON...
+// types to type-fest.
+import type { JsonArray, JsonObject, JsonPrimitive, JsonValue } from 'type-fest'
+
 import HttpError from '@/errors/http'
 
 export type IHttpClient = AxiosInstance
 import type { Request } from 'express'
 
-export type IJSONPrimitive = string | number | boolean
-export type IJSONValue = IJSONPrimitive | IJSONObject | IJSONArray
-export type IJSONArray = Array<IJSONValue>
-export interface IJSONObject {
+export type IJSONPrimitive = string | number | boolean | JsonPrimitive
+export type IJSONValue = IJSONPrimitive | IJSONObject | IJSONArray | JsonValue
+export type IJSONArray = Array<IJSONValue> | JsonArray
+export interface IJSONObject extends JsonObject {
```

Comment:
noob question: isn't the Json types from type-fest meant to replace the ones before e.g. `JsonPrimitive` replaces `string | number | boolean`, is it kept there as a safeguard in case something breaks?

Reply:
good call. shortened the types